soc/xtensa/intel_adsp/ace_v1x/ace-ipc-regs.h

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2021 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_SOC_INTEL_ADSP_ACE_IPC_REGS_H
+#define ZEPHYR_SOC_INTEL_ADSP_ACE_IPC_REGS_H
+
+/* Inter Processor Communication: a distressingly heavyweight method
```

Comment:
This file has passed PR in out internal process and this is not the place to talk about things that are integrated already.
This is upstream, dropping sorry,

Reply:
Andy IPC/IDC PR is still in review in the innersource repos and maintainers are encouraging me to rename driver from cavs_ipc to more common name. It makes sense as cAVS and ACE despite their similarities are different SoCs.
I would like to leave this header as it currently is. So we would have something like intel_adsp_ipc driver and different headers for ACE and CAVS IPC registers. To further support my claim, here are arguments for doing so:

1. Splitting struct cavs_ipc for another \#ifdef MTL "case" is not elegant. They are not that similar due to "memory gaps" in MTL registers and would look awful and confusing. Like so:
![image](https://user-images.githubusercontent.com/12482515/174908699-e7afe70b-b54e-46d9-8827-0def34e250f4.png)

1. Putting registers of one SoC family into registers of other SoC family is simply wrong despite its similarities.
2. Why ACE code should know what this definition is:
`#define CAVS_IPC_IDD15_DONE BIT(30) /* v1.5 way to signal done */`

1. Why would CAVS code should know what this definition is:
`#define ACE_IPC_TDA1X_DONE BIT(0)`
(this one is not present here it is under review on innersource but required by ACE).

So if you like objectively look at those two headers and compare their content you will see that like 50% is different.
It is a lot (skipping argumentation that new SoC should have separate definitions).

Reply:
I'm not seeing your point.  The differences between the two layouts are that one register (CAP) was added to the end of the block and two registers (IDD and TDD) moved to a spot a few dozen words later in memory.  That seems like pretty straightforward migration churn to me and not worth a separate API.

But really the concern is more: there's a seven[1] paragraph comment here explaining how this interface works, which is clearly applicable to both cAVS and MTL/ACE because it's repeated verbatim in both places.  And if we have an interface so complicated that it requires seven paragraphs to explain and understand[2], we really don't want to be maintaining two different drivers for it.

[1] Good grief, the author was a windbag.

[2] Which, OK, the author is famously a windbag and maybe this isn't so necessary.